app/Http/Controllers/Backend/PermissionsController.php

```diff
@@ -0,0 +1,253 @@
+<?php
+namespace App\Http\Controllers\Backend;
+
+use Illuminate\Http\Request;
+use App\Http\Requests;
+use App\Http\Controllers\Controller;
+use App\Models\Backend\Permission;
+
+class PermissionsController extends Controller
+{
+    /**
+     * Display a listing of the Permissions.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $data = Permission::all();
+        return view('backend.permissions.index')->with('data', $data);
+    }
+
+    /**
+     * Show the form for creating a new Permissions.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $modules = $this->getModule();
+        return view('backend.permissions.create')->with('modules', $modules);
+    }
+
+    /**
+     * Store a newly created Permissions in storage.
+     *
+     * @param \Illuminate\Http\Request $request request create
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        // check permission is exists?
+        if (!$this->checkExists($request)) {
+            // create new permission
+            $newRole = new Permission;
+            $newRole->module = $request->module;
+            
+            // set permission
+            $arrRole = $this->setPermission($request);
+            $newRole->see     = $arrRole['see'];
+            $newRole->addNew  = $arrRole['addNew'];
+            $newRole->edit    = $arrRole['edit'];
+            $newRole->destroy = $arrRole['destroy'];
+
+            // check save successfull?
+            if ($newRole->save()) {
+                $request->session()->flash('message', trans('labels.MessageAddRoleOk'));
+            } else {
+                $request->session()->flash('message', trans('labels.MessageAddRoleNotOk'));
+            }
+        } else {
+            $request->session()->flash('message', trans('labels.MessageRoleExists'));
+        }
+        return redirect()->route('admin.permissions.index');
+    }
+
+    /**
+     * Show the form for editing the specified Permissions.
+     *
+     * @param int $id id
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $modules = $this->getModule();
+        $permission = Permission::findOrFail($id);
+        return view('backend.permissions.edit')->with(['modules' => $modules, 'permission' => $permission]);
```

Comment:
[rank B] why don't you use $data["modules"] and $data["permissions"]? And when you are returning with view, it's should be ....->($data)
In the view, we can get $modules and $permissions to handling